src/shared/components/post/post-form.tsx

```diff
@@ -155,49 +148,39 @@ export class PostForm extends Component<PostFormProps, PostFormState> {
     }
   }
 
-  componentDidUpdate() {
-    if (
-      !this.state.loading &&
-      (this.state.form.name || this.state.form.url || this.state.form.body)
-    ) {
-      window.onbeforeunload = () => true;
-    } else {
-      window.onbeforeunload = null;
-    }
-  }
-
-  componentWillUnmount() {
-    this.subscription?.unsubscribe();
-    /* this.choices && this.choices.destroy(); */
-    window.onbeforeunload = null;
-  }
-
   static getDerivedStateFromProps(
-    { selectedCommunityChoice }: PostFormProps,
+    { selectedCommunityChoice, post_view }: PostFormProps,
```

Comment:
It would be nice if none of this were needed.

Reply:
Resolved from changes merged from `use_http_client_dess`.